lib/matplotlib/backends/backend_pgf.py

```diff
@@ -265,7 +267,9 @@ def _expect(self, s):
             if buf[-len(exp):] == exp:
                 break
             if not len(b):
-                raise LatexError("LaTeX process halted", buf.decode("utf8"))
+                self.latex.kill()
+                self.latex = None
+                raise LatexError("LaTeX process halted", "".join(b))
```

Comment:
This can't be right, since the condition says that `b` is empty?
```suggestion
                raise LatexError("LaTeX process halted", "".join(buf))
```